src/equations/heat_equation_2d.jl

```diff
@@ -0,0 +1,90 @@
+
+@doc raw"""
+    HeatEquation2D
+
+The linear scalar advection equation
+```math
+\partial_t u + a_1 \partial_1 u + a_2 \partial_2 u = 0
+```
+in two space dimensions with constant velocity `a`.
+"""
+struct HeatEquation2D{RealT<:Real} <: AbstractHeatEquation{2, 1}
+  nu::RealT
+end
+
+get_name(::HeatEquation2D) = "HeatEquation2D"
+varnames(::typeof(cons2cons), ::HeatEquation2D) = SVector("scalar")
+varnames(::typeof(cons2prim), ::HeatEquation2D) = SVector("scalar")
+
+
+# Set initial conditions at physical location `x` for time `t`
+"""
+    initial_condition_constant(x, t, equations::HeatEquation2D)
+
+A constant initial condition to test free-stream preservation.
+"""
+function initial_condition_constant(x, t, equation::HeatEquation2D)
+  return @SVector [2.0]
+end
+
+
+"""
+    initial_condition_convergence_test(x, t, equations::HeatEquation2D)
+
+A smooth initial condition used for convergence tests.
+"""
+function initial_condition_convergence_test(x, t, equation::HeatEquation2D)
+  @unpack nu = equation
+  c = 1.0
+  A = 0.5
+  L = 2
+  f = 1/L
+  omega = 2 * pi * f
+  scalar = c + A * sin(omega * sum(x)) * exp(-2 * nu * omega^2 * t)
+  return @SVector [scalar]
+end
+
+
+"""
+    initial_condition_gauss(x, t, equation::HeatEquation2D)
+
+A Gaussian pulse used together with
+[`boundary_condition_gauss`](@ref).
+"""
+function initial_condition_gauss(x, t, equation::HeatEquation2D)
+  return @SVector [exp(-(x[1]^2 + x[2]^2))]
+end
+
+
+# Pre-defined source terms should be implemented as
+# function source_terms_WHATEVER(u, x, t, equations::HeatEquation2D)
+
+
+# Calculate parabolic 1D flux in axis `orientation` for a single point
+@inline function calcflux(u, gradients, orientation, equation::HeatEquation2D)
+  return -equation.nu*gradients[orientation]
+end
+
+
+@inline have_constant_speed(::HeatEquation2D) = Val(true)
+
+@inline function max_abs_speeds(equation::HeatEquation2D)
+  return NaN
+end
```

Comment:
Would it be better not to define this function such that you get a method error when trying to call it?

Reply:
Done in 7716534b